### PR TITLE
Correction of depth measures

### PR DESCRIPTION
**Pull request by BenjaminHug8**
**Description**
Related to https://github.com/cyberbotics/urdf2webots/pull/161#issuecomment-1055116459, the RangeFinder measures were not correct close to the near plane.

This PR change the way how the depth is measured.

### CODE REVIEW

**Comment by BenjaminHug8** [comment]
I have to check if we can remove some parts of https://github.com/cyberbotics/webots/pull/4167 used when we get `Nan` instead of `Inf`.

**Comment by BenjaminHug8** [comment]
> I have to check if we can remove some parts of #4167 used when we get `Nan` instead of `Inf`.

We still need the check for `Nan` in the motion blur shader.

**Review by omichel — CHANGES_REQUESTED** [review]
Thank you. I would however add an entry in the change log about it as it is a significant bug fix.

File: docs/reference/changelog-r2022.md [hunk 1]
```diff
@@ -16,13 +16,14 @@ Released on XX XX, 2022.
...
     - Converted missing sample world `gears.wbt` to ENU ([4201](https://github.com/cyberbotics/webots/pull/4201)).
     - Fixed texture of [Camera](camera.md) devices not being retrieved ([#4218](https://github.com/cyberbotics/webots/pull/4218)).
     - Fixed bug where changes in a DEF node did not propagate for PROTO ([#4245](https://github.com/cyberbotics/webots/pull/4245)).
     - Fixed incorrect update of [Mesh](mesh.md) node in a [Shape](shape.md) when the url is updated either manually or from a supervisor ([#4245](https://github.com/cyberbotics/webots/pull/4245)).
     - Fixed a bug that caused an object to sink into the ground after moving it with a supervisor ([#4070](https://github.com/cyberbotics/webots/pull/4070)).
     - Fixed bug where the [Skin](skin.md) node was invisible both to segmentation and `RangeFinder` devices ([#4281](https://github.com/cyberbotics/webots/pull/4281)).
+    - Fixed measures close to the near plane for the [RangeFinder](rangefinder.md) device ([#4309](https://github.com/cyberbotics/webots/pull/4309)).
```
Comment (omichel) [inline docs/reference/changelog-r2022.md:26]:
```suggestion
    - Fixed measurements close to the near plane for the [RangeFinder](rangefinder.md) device ([#4309](https://github.com/cyberbotics/webots/pull/4309)).
```

**Review by omichel — CHANGES_REQUESTED** [review]

**Review by omichel — APPROVED** [review]
Thank you.

**Pull request merged**